zkvm/src/prover.rs

```diff
@@ -57,10 +56,10 @@ impl<'a, 'b> Delegate<r1cs::Prover<'a, 'b>> for Prover<'a, 'b> {
         Ok(run.program.pop_front())
     }
 
-    fn new_run(&self, data: Data) -> Result<Self::RunType, VMError> {
-        Ok(ProverRun {
-            program: data.to_program()?.to_vec().into(),
-        })
+    fn new_run(&self, program: Program) -> Self::RunType {
```

Comment:
Is this change necessary? The verifier intentionally takes raw bytes so it can parse the program on the fly. Otherwise, it'd have to allocate 20x or something memory to store `Vec<Instruction>`

Reply:
my thought was that it would be nice to have the process fail earlier if the `Data` that we pop off the stack is not program (rather than having it fail in `new_run`?) but not much reason besides that

Reply:
That won’t happen on verifier anyway - it has opaque data, but it will on the prover - we can mandate that the program is not opaque there if we want.